.grit/patterns/rust/collapsible_if.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Combine the conditions of nested if statements
+---
+
+Redundant layers of nesting add undesirable complexity.
+
+tags: #clippy
+
+```grit
+language rust
+
+if_expression($condition, $consequence) where {
+    $consequence <: block($content) where {
+        $length = length($content),
+        $length <: 1,
```

Comment:
This length check shouldn't be necessary, since line 16 matches that it's a list of length 1.